src/service/mercury/helpers/transformers.ts

```diff
@@ -145,4 +147,941 @@ const transformAccountBalances = async (
+        sourceMuxed: string;
+      };
+    }[];
+  };
+  setTrustLineFlagsByPublicKey: {
+    edges: {
+      node: {
+        asset: string;
+        assetNative: boolean;
+        clearFlags: boolean;
+        setFlags: boolean;
+        source: string;
+        sourceMuxed: string;
+        trustor: string;
+      };
+    }[];
+  };
+  liquidityPoolDepositByPublicKey: {
+    edges: {
+      node: {
+        maxAmountA: string;
+        maxAmountB: string;
+        maxPriceD: string;
+        maxPriceN: string;
+        minPriceD: string;
+        source: string;
+        sourceMuxed: string;
+      };
+    }[];
+  };
+  liquidityPoolWithdrawByPublicKey: {
+    edges: {
+      node: {
+        amount: string;
+        minAmountA: string;
+        minAmountB: string;
+        source: string;
+        sourceMuxed: string;
+      };
+    }[];
+  };
+}
+
+const transformAccountHistory = async (
+  rawResponse: OperationResult<MercuryAccountHistory>,
+  pubKey: string,
+  network: NetworkNames,
+  getTokenDetails: any // todo
+): Promise<Partial<Horizon.ServerApi.OperationRecord>[]> => {
+  const transferFromEdges = rawResponse.data?.transferFromEvent.edges || [];
+  const transferFrom = await Promise.all(
+    transferFromEdges.map(async (edge) => {
+      const tokenDetails = await getTokenDetails(
+        pubKey,
+        edge.node.contractId,
+        network
+      );
+      const amountBigInt = scValToNative(
+        xdr.ScVal.fromXDR(edge.node.data, "base64")
+      ) as BigInt;
+      return {
+        source_account: pubKey,
+        asset_issuer: edge.node.contractId,
+        asset_code: tokenDetails.symbol,
+        type: "invoke_host_function",
+        type_i: 24,
+        contractId: edge.node.contractId,
+        fnName: scValToNative(xdr.ScVal.fromXDR(edge.node.topic1, "base64")),
+        to: scValToNative(xdr.ScVal.fromXDR(edge.node.topic3, "base64")),
+        from: scValToNative(xdr.ScVal.fromXDR(edge.node.topic2, "base64")),
+        amount: amountBigInt.toString(),
+      } as Partial<Horizon.ServerApi.InvokeHostFunctionOperationRecord>;
+    })
+  );
+
+  const transferToEdges = rawResponse.data?.transferToEvent.edges || [];
+  const transferTo = await Promise.all(
+    transferToEdges.map(async (edge) => {
+      const tokenDetails = await getTokenDetails(
+        pubKey,
+        edge.node.contractId,
+        network
+      );
+      const amountBigInt = scValToNative(
+        xdr.ScVal.fromXDR(edge.node.data, "base64")
+      ) as BigInt;
+      return {
+        source_account: pubKey,
+        asset_issuer: edge.node.contractId,
+        asset_code: tokenDetails.symbol,
+        type: "invoke_host_function",
+        type_i: 24,
+        contractId: edge.node.contractId,
+        fnName: scValToNative(xdr.ScVal.fromXDR(edge.node.topic1, "base64")),
+        to: scValToNative(xdr.ScVal.fromXDR(edge.node.topic3, "base64")),
+        from: scValToNative(xdr.ScVal.fromXDR(edge.node.topic2, "base64")),
```

Comment:
there's some repetition in the return value that can be cleaned up in a future PR:

For ex: you could:
```
const fnName = scValToNative(xdr.ScVal.fromXDR(edge.node.topic1, "base64"));

```

Reply:
Hmm, I don't see where? Each `scValToNative(xdr.ScVal.fromXDR(edge.node.topic1, "base64"))` is for a different event in this case.

Reply:
Oh my bad - I read this wrong. Disregard my code suggestion above :) But I do wonder if some helpers could be useful here. Maybe something more like
```
const transferEdgeHelper = (edge) => {
        contractId: edge.node.contractId,
        fnName: scValToNative(xdr.ScVal.fromXDR(edge.node.topic1, "base64")),
        to: scValToNative(xdr.ScVal.fromXDR(edge.node.topic3, "base64")),
        from: scValToNative(xdr.ScVal.fromXDR(edge.node.topic2, "base64")),
}
```

Mostly just trying to guard against these attrs from falling out of sync because of a typo

Reply:
oh yeah 100%, I'll make that helper since they're all the same shape.

Reply:
hmm actually after looking at this again I think the differences make this trickier to abstract. Each topic from each event correspond to a different key dependent on the event. So even though we see a few lines that have similar code like `scValToNative(xdr.ScVal.fromXDR(some topic, "base64"))`, they name of the key and topic should be mapped according to what the event itself names them, so for mint - topic 2 is the `to` arg but for transferTo, the `to` arg is topic3. 

Let me know if you think differently but I think this means the helper would have to be per topic which seems like less of an abstraction.